### PR TITLE
Use mixlib-install to resolve available Delivery versions

### PR DESCRIPTION
**Pull request by schisamo**
/cc @dmccown @stephenlauck @afiune @jonsmorrow @sersut @patrick-wright

### CODE REVIEW

**Comment by schisamo** [comment]
Looks like Travis isn't properly configured for this repo (I don't even see a `.travis.yml`).

File: libraries/helpers.rb [hunk 1]
```diff
@@ -32,22 +32,15 @@ def add_all_change_data_to_node(node)
...
-      ## each to get the version with '-1'. We should never have anything other
-      ## than '-1' so we are encuring a bit of calculated risk here for the sake
-      ## of not having to call multiple apis.
-      client.get('/api/search/versions',
-        repos: 'omnibus-stable-local',
-        g: 'com.getchef',
-        a: 'delivery')['results'].map { |e| "#{e["version"]}-1" }
+      require 'mixlib/install'
+
+      # NOTE: This is using a mixlib-install private API as we still need
```
Comment (wrightp) [inline libraries/helpers.rb:37]: Is the Bintray class missing anything to make this doable?  Let me know and I can add whatever you need.

File: libraries/helpers.rb [hunk 2]
```diff
@@ -32,22 +32,15 @@ def add_all_change_data_to_node(node)
...
-      ## returns the version as 0.3.73 in the outer versions call even though
-      ## the artifact is 0.3.37-1. We'd have to make an additional call for
-      ## each to get the version with '-1'. We should never have anything other
-      ## than '-1' so we are encuring a bit of calculated risk here for the sake
-      ## of not having to call multiple apis.
-      client.get('/api/search/versions',
-        repos: 'omnibus-stable-local',
-        g: 'com.getchef',
-        a: 'delivery')['results'].map { |e| "#{e["version"]}-1" }
+      require 'mixlib/install'
```
Comment (wrightp) [inline libraries/helpers.rb:35]: You will have to require the full path right now (`mixlib/install/backend/bintray`) to load `Backend::Bintray`

Comment (schisamo) [inline libraries/helpers.rb:35]: @patrick-wright not anymore, I snuck this in a while back:
https://github.com/chef/mixlib-install/pull/74

:smile:

Comment (wrightp) [inline libraries/helpers.rb:35]: brilliant!

**Comment by stephenlauck** [comment]
👍

**Comment by schisamo** [comment]
Since these Travis failures are a red herring I tweaked turned on _Build only if .travis.yml is present_ for the repo.

**Pull request merged**